### PR TITLE
Feature/detailed priors on grids

### PR DESCRIPTION
**Pull request by adolgert**
This enables a modeler to

* select mulstd priors on the Rate, Random Effect, Study Covariate, and Country Covariate tabs
* create detailed priors on the Rate, Random Effect, Study Covariate, and Country Covariate tabs

The second feature is more likely to be useful.

### CODE REVIEW

File: src/cascade/executor/construct_model.py [hunk 1]
```diff
@@ -147,16 +148,65 @@ def constraint_from_rectangular_data(rate_var, default_age_time):
...
+        smoothing_prior (cascade.input_data.configuration.form.SmoothingPrior):
+            A single smoothing prior.
+
+    Returns:
+        Iterator over (a, t) that match. Can be nothing.
+    """
+    extents = dict()
+    for extent in ["age", "time", "born"]:
+        extents[extent] = np.zeros(2, dtype=np.float)
+        for sidx, side, default_extent in [(0, "lower", -inf), (1, "upper", inf)]:
```
Comment (alec-deason) [inline src/cascade/executor/construct_model.py]: I stumbled over `sidx`. I think `s_idx` or `side_idx` would have been clear to me.

File: src/cascade/executor/construct_model.py [hunk 2]
```diff
@@ -147,16 +148,65 @@ def constraint_from_rectangular_data(rate_var, default_age_time):
...
+            name = f"{extent}_{side}"
+            if smoothing_prior.is_field_unset(name):
+                extents[extent][sidx] = default_extent
+            else:
+                extents[extent][sidx] = getattr(smoothing_prior, name)
+    ages, times = np.meshgrid(rate_grid.ages, rate_grid.times)
+    # result is shape (len(times), len(ages)), backwards
```
Comment (alec-deason) [inline src/cascade/executor/construct_model.py]: I have no idea what this means

File: src/cascade/executor/construct_model.py (hunk 1 above)
Comment (adolgert) [reply src/cascade/executor/construct_model.py]: That helps.

File: src/cascade/executor/construct_model.py (hunk 2 above)
Comment (adolgert) [reply src/cascade/executor/construct_model.py]: That helps, too. I can express this more clearly. This will be read more times than it was written.

**Pull request merged**